core/dataflow/src/main/java/edu/uci/ics/texera/dataflow/udf/PythonUDFOperator.java

```diff
@@ -0,0 +1,700 @@
+                                TimeSecVector subVectorTime = (TimeSecVector) ((StructVector) vector).getChildByOrdinal(1);
+                                texeraField = new DateTimeField(
+                                        LocalDateTime.of(
+                                                LocalDate.ofEpochDay(subVectorDay.get(i)),
+                                                LocalTime.ofSecondOfDay(subVectorTime.get(i))
+                                        )
+                                );
+                            } catch (ClassCastException e) {
+                                TimeMicroVector subVectorTime = (TimeMicroVector) ((StructVector) vector).getChildByOrdinal(1);
+                                texeraField = new DateTimeField(
+                                        LocalDateTime.of(
+                                                LocalDate.ofEpochDay(subVectorDay.get(i)),
+                                                LocalTime.ofSecondOfDay(subVectorTime.get(i)/1000000)
+                                        )
+                                );
+                            }
+                            break;
+                        case List:
+                            texeraField = getSpanFromListVector((ListVector) vector, i);
+                            break;
+                        default:
+                            schemaRoot.clear();
+                            closeClientAndServer();
+                            throw (new DataflowException("Unsupported data type "+
+                                    vector.getField().toString() +
+                                    " when converting back to Texera table."));
+                    }
+                } catch (IllegalStateException | IOException e) {
+                    if (!e.getMessage().contains("Value at index is null")) {
+                        schemaRoot.clear();
+                        closeClientAndServer();
+                        throw new DataflowException(e.getMessage(), e);
+                    } else {
+                        switch (vector.getField().getFieldType().getType().getTypeID()) {
+                            case Int: texeraField = new IntegerField(null); break;
+                            case FloatingPoint: texeraField = new DoubleField(null); break;
+                            case Date: texeraField = new DateField((String) null); break;
+                            case Struct: texeraField = new DateTimeField((String) null); break;
+                            case List: texeraField = new ListField<Span>(null);
+                            default: break;
+                        }
+                    }
+                }
+                texeraFields.add(texeraField);
+            }
+            tuple = new Tuple(texeraSchema, texeraFields);
+            resultQueue.add(tuple);
+        }
+    }
+
+    private Schema convertToTexeraSchema(org.apache.arrow.vector.types.pojo.Schema arrowSchema) {
```

Comment:
if possible, make util functions like this `private static`

Reply:
OK.

Reply:
Done. All util functions are made static.